lib/rubocop/cop/lint/nested_method_definition.rb

```diff
@@ -55,17 +55,17 @@ module Lint
       #     end
       #   end
       class NestedMethodDefinition < Cop
-        include OnMethodDef
-        extend RuboCop::NodePattern::Macros
-
         MSG = 'Method definitions must not be nested. ' \
               'Use `lambda` instead.'.freeze
 
-        def on_method_def(node, _method_name, _args, _body)
+        def on_def(node)
           find_nested_defs(node) do |nested_def_node|
             add_offense(nested_def_node)
           end
         end
+        alias on_defs on_def
```

Comment:
I'm planning to organize the custom callbacks somewhere. There are also a couple other ones that would be nice to abstract out, like `#on_heredoc`.